spring-cloud-dataflow-aggregate-task/src/main/java/org/springframework/cloud/dataflow/aggregate/task/impl/AggregateDataFlowTaskExecutionQueryDao.java

```diff
@@ -509,15 +539,19 @@ private Page<AggregateTaskExecution> queryForPageableResults(
 		}
 		String query = pagingQueryProvider.getPageQuery(pageable);
 		List<AggregateTaskExecution> resultList = this.jdbcTemplate.query(query,
-				queryParameters, new CompositeTaskExecutionRowMapper());
+				queryParameters, new CompositeTaskExecutionRowMapper(false));
+		resultList.stream()
```

Comment:
If these are improvements that are not related to the thin controller effort then I think it would be good to move them into the subsequent PR for "Improve performance of executions queries/APIs".

Reply:
The row mapper retrieved arguments using a subquery. The arguments are not part of response needed for `tasks/thinexecutions`